### ISSUE TITLE
False positive in "Statements should be on separate lines"

### ISSUE THREAD

**Issue by wichtounet** [wenns/sonar-cxx#445]
I have some false positives with the rule "Statements should be on separate lines". 

I have this on using statements inside functions: 

```c++
using type = typename A::value_type;
```
with this message: At most one statement is allowed per line, but 2 statements were found on this line.

I also have false positives on statements containing a lambda function: 

```c++
return std::find_if(begin(expr), end(expr), [](auto& v){return !std::isfinite(v);}) == end(expr);
```

but in this case, depending on personal taste, some people may think that this is a true positive.

**Comment by guwirth** [comment]
@Bertk @Typz could one of you fix this?

**Comment by guwirth** [comment]
@wichtounet See #451. Does this fix your problem?

**Comment by guwirth** [comment]
Fixed with #451

**Comment by wichtounet** [comment]
I finally took the time to install the last version of the plugin. The last commit of the master. 

Unfortunately, I still have several of these false positives in my analysis. It seems that there is less issues than before, but I still have 63 of these. Maybe it is simply a coincidence but they all appear in template functions.

**Comment by wichtounet** [comment]
I should also mention that I have several other false positives on lines with using statements: 

using type = etl::fast_vector<ZZZ, 4>;
Redundant code: Found a statement that begins with numeric constant. 

using IT = intrinsic_traits<value_t<E>>;
Comparison of a boolean value using relational operator (<, >, <= or >=).

**Comment by LegalizeAdulthood** [comment]
`*cough*` clang libtooling

**Comment by Bertk** [comment]
Please use https://github.com/wenns/sonar-cxx/issues/507 to report this kind of issues. The plugin parser (/cxx-squid/src/main/java/org/sonar/cxx/parser/CxxGrammarImpl.java) need to be improved for the C++11 alias declarations (http://en.cppreference.com/w/cpp/language/type_alias).
The statement check shows only the symptom of a parser problem and you can identify this parser issues with ‘C++ parser failure’ (rule: cxx:ParsingError) or ‘C++ skip parser error’ (rule: cxx:ParsingErrorRecovery).
This is a valuable input for the C++11 support and we will need to add some C++14/C++17 features as well.

**Comment by Bertk** [comment]
@wichtounet Please provide a source code example for every single issues and create new issues if we do not have one open which has the identical root cause.
